### PR TITLE
Fixes to report CLI and to/from dict

### LINKED ISSUE #137: error when dumping Report data model object to_dict
Calling `to_dict` on the report data model object returned from the API results in the following error when it hits the `defaultdict` of metrics/
TypeError: first argument must be callable or None

### LINKED ISSUE #236: report CLI fxobs ordering reversed
compare https://github.com/SolarArbiter/solarforecastarbiter-core/blob/1e0d2d4f494c5819635cbed101739515726341aa/solarforecastarbiter/cli.py#L311-L312
to 
https://github.com/SolarArbiter/solarforecastarbiter-core/blob/1e0d2d4f494c5819635cbed101739515726341aa/solarforecastarbiter/io/api.py#L588-L591

### PR DESCRIPTION
**Pull request by alorenzo175**
- [x] Closes #137 closes #236 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: docs/source/whatsnew/1.0.0b2.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .. _whatsnew_100b2:
 
-1.0.0b2 (???)
+1.0.0b2 (November 15, 2019)
```
Comment (wholmgren) [inline docs/source/whatsnew/1.0.0b2.rst]: I think the --- below is supposed to have the same number of characters as this line

File: solarforecastarbiter/datamodel.py [hunk 2]
```diff
@@ -784,20 +816,6 @@ def __post_init__(self):
         __set_constant_values__(self)
         __check_axis_consistency__(self.axis, self.constant_values)
 
-    @classmethod
-    def from_dict(model, input_dict, raise_on_extra=False):
```
Comment (wholmgren) [inline solarforecastarbiter/datamodel.py]: changes to `_special_field_processing` make this unnecessary? I have no idea what's going on up there.

File: docs/source/whatsnew/1.0.0b2.rst [hunk 3]
```diff
@@ -52,7 +55,10 @@ Bug fixes
   (:issue:`204`)
 * :py:class:`datamodel.Report` and associated classes were missing from
   :ref:`api` documentation. Fixed. (:issue:`228`)
-
+* Fix loading :py:class:`datamodel.Report` and :py:class:`datamodel.BaseFilter`
+  from a dictionary (:pull:`249`)
+* :py:class:`io.api.APISession` Report methods where  missing from
```
Comment (wholmgren) [inline docs/source/whatsnew/1.0.0b2.rst]:
```suggestion
* :py:class:`io.api.APISession` Report methods were missing from
```

File: docs/source/whatsnew/1.0.0b2.rst [hunk 4]
```diff
@@ -52,7 +55,10 @@ Bug fixes
   (:issue:`204`)
 * :py:class:`datamodel.Report` and associated classes were missing from
   :ref:`api` documentation. Fixed. (:issue:`228`)
-
+* Fix loading :py:class:`datamodel.Report` and :py:class:`datamodel.BaseFilter`
+  from a dictionary (:pull:`249`)
```
Comment (wholmgren) [inline docs/source/whatsnew/1.0.0b2.rst]:
```suggestion
  from a dictionary (:pull:`249`, :issue:`137`)
```

**Comment by wholmgren** [comment]
are we ok with the missing coverage? https://codecov.io/gh/SolarArbiter/solarforecastarbiter-core/compare/a1002b162cf991afeaca87353d2aa08920212c53...821a13e83d5d4df7ed27273ad28824f569b1fcfc/diff#D4-895...899

File: solarforecastarbiter/datamodel.py (hunk 2 above)
Comment (alorenzo175) [reply solarforecastarbiter/datamodel.py]: changes to handling Union in _single_field_processing makes it unnecessary

**Comment by alorenzo175** [comment]
I'm ok with missing coverage. Only the quality flag filters even do anything right now.

**Comment by wholmgren** [comment]
@alorenzo175 ok to merge if you're ready

**Pull request merged**